subprojects/core/src/main/java/org/gradle/invocation/DefaultGradle.java

```diff
@@ -183,8 +183,7 @@ public String getGradleVersion() {
 
     @Override
     public File getGradleHomeDir() {
```

Comment:
I think this changes the meaning of `gradle.gradleHomeDir` a bit...

> Returns the Gradle home directory, if any. This directory is the directory containing the Gradle distribution executing this build.
> When using the “Gradle Daemon”, this may not be the same Gradle distribution that the build was started with. If an existing daemon process is running that is deemed compatible (e.g. has the desired JVM characteristics) then this daemon may be used instead of starting a new process and it may have been started from a different “gradle home”. 

I read that to mean that `gradle.gradleHomeDir` should always be the installation of the Gradle daemon that's running the build and not the client requested Gradle installation.

The TAPI failing tests point to that too:
https://builds.gradle.org/viewLog.html?buildId=12814254&tab=buildResultsDiv&buildTypeId=Gradle_Check_Quick_Java7_Oracle_Windows_toolingApi

This is saying that the first build starts a daemon with distro1.  The second build, using distro2, uses the distro1 daemon.  The build sees the init scripts from the right distribution, but `gradle.gradleHomeDir` is the installation directory of the daemon running the build.

Reply:
Makes sense. It just shows that the init script integ test was never working.